go/teams/multiple_use_invite_test.go

```diff
@@ -88,7 +88,117 @@ func TestTeamInviteStubbing(t *testing.T) {
 
 	inner = teamObj.chain().inner
 	require.Len(t, inner.ActiveInvites, 1)
-	_, ok := inner.ActiveInvites[inviteID]
+	invite, ok := inner.ActiveInvites[inviteID]
 	require.True(t, ok, "invite found loaded by user 2")
 	require.Len(t, inner.UsedInvites[inviteID], 1)
+
+	// See if User 2 can decrypt
+	pkey, err := SeitanDecodePKey(string(invite.Name))
+	require.NoError(t, err)
+
+	keyAndLabel, err := pkey.DecryptKeyAndLabel(context.TODO(), teamObj)
+	require.NoError(t, err)
+
+	ilink := keyAndLabel.Invitelink()
+	require.Equal(t, inviteLink.Ikey, ilink.I)
+}
+
+func TestSeitanHandleExceededInvite(t *testing.T) {
+	// Test what happens if server sends us acceptance for an invite that's
+	// exceeded. Handler should notice that and not add the member. Even it it
+	// attempted to, there are additional belts and suspenders:
+
+	// 1) sigchain pre-check should fail,
+	// 2) server should not accept the link,
+	// 3) if none of the above checks worked: the team would have ended up
+	//    broken (not loadable) for other admins.
+
+	tc := SetupTest(t, "team", 1)
+	defer tc.Cleanup()
+
+	tc.Tp.SkipSendingSystemChatMessages = true
+
+	clock := clockwork.NewFakeClockAt(time.Now())
+	tc.G.SetClock(clock)
+
+	user2, err := kbtest.CreateAndSignupFakeUser("team", tc.G)
+	require.NoError(t, err)
+	kbtest.Logout(tc)
+
+	admin, err := kbtest.CreateAndSignupFakeUser("team", tc.G)
+	require.NoError(t, err)
+
+	teamName, teamID := createTeam2(tc)
+	t.Logf("Created team %s", teamName)
+
+	// Add team invite link with max_uses=1
+	maxUses := keybase1.TeamInviteMaxUses(1)
+	invLink, err := CreateInvitelink(tc.MetaContext(), teamName.String(), keybase1.TeamRole_READER, maxUses, nil /* etime */)
+	require.NoError(t, err)
+
+	// Accept the link as user2.
+	kbtest.LogoutAndLoginAs(tc, user2)
+
+	sikey, err := GenerateSIKeyInvitelink(invLink.Ikey)
+	require.NoError(t, err)
+	inviteID, err := sikey.GenerateTeamInviteID()
+	require.NoError(t, err)
+
+	now := clock.Now()
+	uv := user2.GetUserVersion()
+	_, encoded, err := GenerateSeitanInvitelinkAcceptanceKey(sikey[:], uv.Uid, uv.EldestSeqno, now.Unix())
+	require.NoError(t, err)
+
+	apiArg := apiArg("team/seitan_invitelink")
```

Comment:
Maybe better to factor out code so you don't need to do this in test

Reply:
yes I have a refactor PR that does that but it's a big lift